torch/csrc/jit/script/compiler.cpp

```diff
@@ -2507,11 +2504,22 @@ struct to_ir {
             loc,
             integral_constants);
         range_inputs.emplace_back(end);
-        SugaredValuePtr range_sv =
-            std::make_shared<RangeValue>(loc, method, range_inputs);
         SugaredValuePtr expr_sv = emitSugaredExpr(inputs[0], 1);
-        return std::make_shared<IterableTree>(
-            std::vector<SugaredValuePtr>({range_sv, expr_sv}));
+        auto iterable_value = expr_sv->iter(loc, method);
+
+        // range should have the same static length as the other iterable
+        c10::optional<int64_t> iter_static_len = c10::nullopt;
```

Comment:
Discussed in person: only ranges created from enumerate need to have static lengths. If we do this then we can making have a static length and choosing to unroll the same decision.